### PR TITLE
Remove Author and Signature fields from Record

### PR DESCRIPTION
**Pull request by dirkmc**

### CODE REVIEW

File: pb/record.proto [hunk 1]
```diff
@@ -16,12 +16,6 @@ message Record {
...
 
-	// hash of the authors public key
-	optional string author = 3;
-
-	// A PKI signature for the key+value+author
-	optional bytes signature = 4;
-
 	// Time the record was received, set by receiver
-	optional string timeReceived = 5;
+	optional string timeReceived = 3;
```
Comment (vyzo) [inline pb/record.proto]: please don't change the field id, it is backwards incompatible.

Comment (Stebalien) [reply pb/record.proto]: (although I'm pretty sure that's *technically* a harmless change.

File: pb/record.proto [hunk 2]
```diff
@@ -16,12 +16,6 @@ message Record {
 	// The actual value this record is storing
 	optional bytes value = 2;
 
-	// hash of the authors public key
-	optional string author = 3;
-
-	// A PKI signature for the key+value+author
-	optional bytes signature = 4;
```
Comment (Stebalien) [inline pb/record.proto]: We should probably leave a comment reserving these fields.

**Comment by vyzo** [comment]
i think @whyrusleeping should chime in here too, that's a pretty drastic change in behaviour and i'd like to know if we'll break anything by suddenly removing signatures and their validation.

File: pb/record.proto (hunk 1 above)
Comment (dirkmc) [reply pb/record.proto]: Nice catch.. please take a look at the diff of `record.pb.go`. The protoc compiler I'm using has added in a field called `Descriptor`. I'm not sure what that is so just want to make sure it's ok

Comment (dirkmc) [reply pb/record.proto]: Never mind I fixed

**Comment by Stebalien** [comment]
@vyzo I've talked with @whyrusleeping offline but yes, I'd like him to give a final signoff. This *shouldn't* break anything as we only validate signatures iff they're present but still...

After talking about this with Juan again (and again), we're pretty sure this signature came about as an early attempt at IPRS. Unfortunately, when we added support for multiple IPNS keys, we kept on signing the DHT records with the wrong key so trying to use this for IPRS at this point wouldn't work all that well...

Personally, I'd almost like to just introduce IPRS as separate path. For the "backwards compat period", ipfs nodes would publish/retrieve both the IPRS and old records. After that compatibility period, they'd stop retrieving the old records. With request pipelining, this shouldn't be that expensive and allows us to start over.

Anyways, the way we do signatures on these records and IPNS records is kind of broken and is only secure by happy coincidence (https://github.com/ipfs/notes/issues/249).

**Comment by Stebalien** [comment]
LGTM. Thanks @dirkmc!

**Comment by dirkmc** [comment]
Glad to be of help
Do you think we still need https://github.com/libp2p/go-libp2p-record/issues/10?

**Comment by Stebalien** [comment]
@dirkmc I'd still like to use something like the `ValidationRecord` elsewhere in our code.

**Pull request merged**